src/transform/import_resolution.rs

```diff
@@ -40,41 +81,52 @@ where
         resolver,
         stack: &mut stack,
         parent: source_file,
+        unresolved: &mut unresolved,
     };
 
     // If an import is resolved, then stack it.
-    let transformed = rt.traverse(
-        &|rt: RichTerm, state: &mut ImportsResolutionState<R>| -> Result<RichTerm, ImportError> {
-            let rt = transform_one(rt, state.resolver, &state.parent)?;
+    let transformed = rt
+        .traverse(
+            &|rt: RichTerm,
+              state: &mut ImportsResolutionState<R>|
+             -> Result<RichTerm, ImportError> {
+                let (rt, err) = transform_one_resillient(rt, state.resolver, &state.parent);
+                if let Some(err) = err {
+                    state.unresolved.push(err);
+                }
 
-            if let Term::ResolvedImport(file_id) = rt.term.as_ref() {
-                state.stack.push(*file_id);
-            }
-            Ok(rt)
-        },
-        &mut state,
-        TraverseOrder::BottomUp,
-    )?;
+                if let Term::ResolvedImport(file_id) = rt.term.as_ref() {
+                    state.stack.push(*file_id);
+                }
+                Ok(rt)
+            },
+            &mut state,
+            TraverseOrder::BottomUp,
+        )
+        // This will always succeed, becuase we always return `Ok(..)`, and
+        // we don't use the `?` operator
+        .unwrap();
 
-    Ok((transformed, stack))
+    (transformed, stack, unresolved)
 }
 
-/// Resolve the import if the term is an unresolved import, or return the term unchanged. As
-/// [`super::share_normal_form::transform_one`], this function is not recursive.
-pub fn transform_one<R>(
+/// Try to resolve an import if the term is an unresolved import, and return
+/// back the a potentially resolved richterm and an import error if it couldn't
+/// be unresolved import could not be resolved.
+pub fn transform_one_resillient<R>(
     rt: RichTerm,
     resolver: &mut R,
     parent: &Option<PathBuf>,
-) -> Result<RichTerm, ImportError>
+) -> (RichTerm, Option<ImportError>)
 where
     R: ImportResolver,
 {
     let term = rt.as_ref();
     match term {
-        Term::Import(path) => {
-            let (_, file_id) = resolver.resolve(path, parent.clone(), &rt.pos)?;
-            Ok(RichTerm::new(Term::ResolvedImport(file_id), rt.pos))
-        }
-        _ => Ok(rt),
+        Term::Import(path) => match resolver.resolve(path, parent.clone(), &rt.pos) {
+            Ok((_, file_id)) => (RichTerm::new(Term::ResolvedImport(file_id), rt.pos), None),
+            Err(err) => (rt, Some(err)),
+        },
+        _ => (rt, None),
     }
 }
```

Comment:
Suggestion: we have several function with similar names, the XXX, and the XXX_resilient. One way to organize this a bit more would be to have submodules, say `strict` and `tolerant` (just because it's the terminology we are already using in the cache; for what it's worth, resilient is totally fine too), each with their own `resolve_imports` and `transform_one` function.